pandas/_libs/period.pyx

```diff
@@ -1178,6 +1179,8 @@ class Period(_Period):
                 value = str(value)
             value = value.upper()
             dt, _, reso = parse_time_string(value, freq)
+            if dt is NAT_SENTINEL:
```

Comment:
why this change?

Reply:
`tslibs.parsing` does not have `NaT` in the namespace, so it returns `NAT_SENTINEL` in places where it otherwise  would return `NaT`.  That should be wrapped in `tslib`, will update.